tests/unit/amo/reducers/test_home.js

```diff
@@ -455,7 +469,31 @@ describe(__filename, () => {
       expect(
         createInternalHeroShelves(heroShelves, lang).secondary,
       ).toMatchObject({
-        modules: heroShelves.secondary.modules,
+        modules: [
+          {
+            icon: heroShelves.secondary.modules[0].icon,
+            description: heroShelves.secondary.modules[0].description[lang],
+            cta: null,
+          },
+          {
+            icon: heroShelves.secondary.modules[1].icon,
+            description: heroShelves.secondary.modules[1].description[lang],
+            cta: {
+              url: heroShelves.secondary.modules[1].cta.url,
+              outgoing: heroShelves.secondary.modules[1].cta.outgoing,
+              text: heroShelves.secondary.modules[1].cta.text[lang],
+            },
+          },
+          {
+            icon: heroShelves.secondary.modules[2].icon,
+            description: heroShelves.secondary.modules[2].description[lang],
+            cta: {
+              url: heroShelves.secondary.modules[2].cta.url,
+              outgoing: heroShelves.secondary.modules[2].cta.outgoing,
+              text: heroShelves.secondary.modules[2].cta.text[lang],
+            },
+          },
+        ],
```

Comment:
```suggestion
        modules: heroShelves.secondary.modules.map((module) => createInternalSecondaryHeroModule(module, lang)),
```

Reply:
for this test, `cta` for the first module is `null` and couldn't immediately think of a way to specify that without writing it all out.

Reply:
Do you really have to? Isn't `createInternalSecondaryHeroModule()` supposed to set `cta` to `null`? Maybe you could keep the `modules` array and override the first entry if it's important:

```js
modules: [
  {
    ...createInternalSecondaryHeroModule(modules[0], lang),
    cta: null,
  },
  createInternalSecondaryHeroModule(modules[1], lang),
  createInternalSecondaryHeroModule(modules[1], lang),
]
```

but what you want is probably some tests for `createInternalSecondaryHeroModule()` (in a separate `describe()` block).

Reply:
yes, that is the drawback of replacing the literal data structure with `createInternalSecondaryHeroModule(module, lang)),` or similar - we're not actually testing the result is as expected any longer.  Unless each of the `createInternal...` functions is also tested seperately  (so likely the same amount of test code, just split over different tests).  I'll add some more `describe` blocks.

Reply:
https://github.com/mozilla/addons-frontend/pull/10086/commits/7d92b37c782685e84da506439bf1f1cf070b5806

Reply:
Possibly some of the testing is duplicated now.  Not sure if it's worth removing it though.